src/integ_test/java/games/strategy/engine/lobby/server/db/AbstractModeratorServiceControllerTestCase.java

```diff
@@ -0,0 +1,107 @@
+package games.strategy.engine.lobby.server.db;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Random;
+import java.util.function.Supplier;
+
+import games.strategy.engine.lobby.server.Moderator;
+import games.strategy.util.Util;
+
+/**
+ * Superclass for fixtures that test a moderator service controller.
+ */
+public abstract class AbstractModeratorServiceControllerTestCase {
+  protected final Moderator moderator = newModerator();
+
+  protected AbstractModeratorServiceControllerTestCase() {}
+
+  /**
+   * Creates a new unique moderator.
+   */
+  protected static Moderator newModerator() {
+    return new Moderator(newUsername(), newInetAddress(), newHashedMacAddress());
+  }
+
+  /**
+   * Creates a new unique username.
+   */
+  protected static String newUsername() {
+    return "user_" + Util.createUniqueTimeStamp();
+  }
+
+  private static InetAddress newInetAddress() {
+    final byte[] addr = new byte[4];
+    new Random().nextBytes(addr);
+    try {
+      return InetAddress.getByAddress(addr);
+    } catch (final UnknownHostException e) {
+      throw new AssertionError("should never happen", e);
+    }
+  }
+
+  /**
+   * Creates a new unique hashed MAC address.
+   */
+  protected static String newHashedMacAddress() {
+    return games.strategy.util.MD5Crypt.crypt(Util.createUniqueTimeStamp(), "MH");
+  }
+
+  /**
+   * Asserts the moderator returned from the specified query is equal to the expected moderator.
+   *
+   * @param expected The expected moderator.
+   * @param moderatorQuerySqlSupplier Supplies the SQL used to query for the moderator. It is expected that this query
+   *        returns the moderator's username in the first column, the moderator's IP address in the second column, and
+   *        the moderator's hashed MAC address in the third column.
+   * @param preparedStatementInitializer Callback to initialize the parameters in the prepared statement used to query
+   *        for the moderator.
+   * @param unknownModeratorMessageSupplier Supplies the failure message to be used when the requested moderator does
+   *        not exist.
+   */
+  protected static void assertModeratorEquals(
+      final Moderator expected,
+      final Supplier<String> moderatorQuerySqlSupplier,
+      final PreparedStatementInitializer preparedStatementInitializer,
+      final Supplier<String> unknownModeratorMessageSupplier) {
```

Comment:
Same here: Do we gain any benefit from this supplier?

Reply:
The only benefit of the lazy evaluation is because the failure message is not actually used in most cases.  However, I agree that it's probably not worth it here and just adds noise.  There is nothing expensive about a compile-time string concatenation at all call sites. :smile:  Updated in rebased ddfa1c9e0.